rs-backend/src/server.rs

```diff
@@ -189,42 +259,37 @@ fn get_data_for_date(day: &TestRun, crate_names: &[String], phases: &[String], g
         rss: u64,
     }
 
+    impl Recording {
+        fn new() -> Recording {
+            Recording {
+                time: 0.0,
+                rss: 0,
+            }
+        }
+
+        fn record(&mut self, phase: Option<&Timing>) {
+            if let Some(phase) = phase {
+                self.time += phase.time;
+                self.rss = max(self.rss, phase.rss.unwrap());
+            }
+        }
+    }
+
     let crates = crate_names.into_iter().filter_map(|crate_name| {
         day.by_crate.get(crate_name).map(|krate| {
             (crate_name, krate)
         })
     }).collect::<Vec<_>>();
 
     let mut data = HashMap::new();
```

Comment:
The new variant of the grouping code is much shorter. However, I'm not sure if it's entirely a win since there is a loss in clarity: instead of inserting after creating we now update in-place, since we always iterate over the same things. The order in which the two for loops are in no longer matters either.